include/vi.h

```diff
@@ -0,0 +1,240 @@
+ */
+typedef struct vi_config_s{
+    vi_register_t regs[VI_REGISTERS_COUNT];
+} vi_config_t;
+
+volatile vi_register_t* VI_REGISTERS    =  ((volatile vi_register_t*)VI_REGISTERS_ADDR);
+/** @brief VI Index register of controlling general display filters/bitdepth configuration */
+volatile vi_register_t* VI_CTRL         =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[0];
+/** @brief VI Index register of RDRAM base address of the video output Frame Buffer. This can be changed as needed to implement double or triple buffering. */
+volatile vi_register_t* VI_ORIGIN       =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[1];
+/** @brief VI Index register of width in pixels of the frame buffer. */
+volatile vi_register_t* VI_WIDTH        =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[2];
+/** @brief VI Index register of vertical interrupt. */
+volatile vi_register_t* VI_V_INTR       =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[3];
+/** @brief VI Index register of the current half line, sampled once per line. */
+volatile vi_register_t* VI_V_CURRENT    =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[4];
+/** @brief VI Index register of sync/burst values */
+volatile vi_register_t* VI_BURST        =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[5];
+/** @brief VI Index register of total visible and non-visible lines. 
+ * This should match either NTSC (non-interlaced: 0x20D, interlaced: 0x20C) or PAL (non-interlaced: 0x271, interlaced: 0x270) */
+volatile vi_register_t* VI_V_SYNC       =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[6];
+/** @brief VI Index register of total width of a line */
+volatile vi_register_t* VI_H_SYNC       =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[7];
+/** @brief VI Index register of an alternate scanline length for one scanline during vsync. */
+volatile vi_register_t* VI_H_SYNC_LEAP  =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[8];
+/** @brief VI Index register of start/end of the active video image, in screen pixels */
+volatile vi_register_t* VI_H_VIDEO      =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[9];
+/** @brief VI Index register of start/end of the active video image, in screen half-lines. */
+volatile vi_register_t* VI_V_VIDEO      =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[10];
+/** @brief VI Index register of start/end of the color burst enable, in half-lines. */
+volatile vi_register_t* VI_V_BURST      =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[11];
+/** @brief VI Index register of horizontal subpixel offset and 1/horizontal scale up factor. */
+volatile vi_register_t* VI_X_SCALE      =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[12];
+/** @brief VI Index register of vertical subpixel offset and 1/vertical scale up factor. */
+volatile vi_register_t* VI_Y_SCALE      =  &((volatile vi_register_t*)VI_REGISTERS_ADDR)[13];
+
+/** @brief VI register by index (0-13)*/
+#define VI_TO_REGISTER(index) ((index >= 0 && index <= VI_REGISTERS_COUNT)? &VI_REGISTERS[index] : NULL)
```

Comment:
Please use parentesis around macro argument usages when used as part of an expression:
```suggestion
#define VI_TO_REGISTER(index) (((index) >= 0 && (index) <= VI_REGISTERS_COUNT)? &VI_REGISTERS[index] : NULL)
```